### PR TITLE
Fix ExprDurability's Changer

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR fixes an issue where if you try to modify an `itemstack`'s durability, it wouldn't properly update its `ItemMeta`.
Also cleans up the class a bit.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/6147

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprDurability.java [hunk 1]
```diff
@@ -89,62 +86,67 @@ public Class<?>[] acceptChange(ChangeMode mode) {
...
-				ItemUtils.setDamage(itemStack, changeValue);
-				((Slot) object).setItem(itemStack);
-			}
+			itemType.setItemMeta(meta);
+			if (object instanceof Slot)
+				((Slot) object).setItem(itemType.getRandom());
 		}
 	}
 
+	private int translate(Material material, int value) {
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprDurability.java:123]: please name this better or give it some comments
it's impossible to tell what it's supposed to do without reading the code inside it

**Review by sovdeeth — APPROVED** [review]
nice work

**Review by Moderocky — APPROVED** [review]
This expression would like some tests :)

File: src/main/java/ch/njol/skript/expressions/ExprDurability.java [hunk 2]
```diff
@@ -89,62 +86,67 @@ public Class<?>[] acceptChange(ChangeMode mode) {
...
 		}
 	}
 
+	private int convertToDamage(Material material, int value) {
+		if (!durability)
+			return value;
+		int maxDurability = material.getMaxDurability();
+		if (maxDurability == 0)
+			return 0;
+		return maxDurability - value;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprDurability.java:129]:
```suggestion
		return Math.max(material.getMaxDurability() - value, 0);
```
Would be nice :) (unless negative values are intended?)

**Review by APickledWalrus — APPROVED** [review]
looks ok

File: src/main/java/ch/njol/skript/expressions/ExprDurability.java (hunk 2 above)
Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/expressions/ExprDurability.java:129]: I am not actually sure if a negative durability value is valid. If they are, then this does not need to be done.

Comment (sovdeeth) [reply src/main/java/ch/njol/skript/expressions/ExprDurability.java:129]: You can certainly have an item with negative durability, it just looks stupid in game
I tested it out when I reviewed this

**Pull request merged**